controllers/commentController.js

```diff
@@ -148,7 +140,161 @@ export default class CommentController {
       }
       res.status(OK).json({ comment });
     } catch (error) {
-      res.status(INTERNAL_SERVER_ERROR).json({ message: "Sorry, this is not working properly. We now know about this mistake and are working to fix it" });
+      res.status(INTERNAL_SERVER_ERROR).json({
+        message:
+          "Sorry, this is not working properly. We now know about this mistake and are working to fix it"
+      });
+    }
+  }
+
+  /**
+   * @description Allows users to fetch a particular comment of an article.
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - It returns the request response object.
+   */
+  static async fetch(req, res) {
+    try {
+      const { id: commentId } = req.params;
+      const comment = await Comment.findOne({ where: { id: commentId } });
+
+      if (!comment) {
+        return res
+          .status(BAD_REQUEST)
+          .json({ message: "The comment does not exist" });
+      }
+      res.status(OK).json({ comment });
+    } catch (error) {
+      res.status(INTERNAL_SERVER_ERROR).json({ message: error.message });
+    }
+  }
+
+  /**
+   * @description Allows users to fetch all comments of an article.
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - It returns the request response object.
+   */
+  static async fetchAllComments(req, res) {
+    try {
+      const { slug } = req.params;
+
+      const article = await Article.findOne({
+        where: { slug },
+        include: [
+          {
+            model: Comment,
+            attributes: ["body"],
+            include: [{ model: User, as: "author", attributes: ["username"] }]
+          }
+        ]
+      });
+
+      if (!article) {
+        return res
+          .status(BAD_REQUEST)
+          .json({ message: "There is no article with that slug" });
+      }
+      res.status(OK).json({ article });
+    } catch (error) {
+      res.status(INTERNAL_SERVER_ERROR).json({ message: error.message });
+    }
+  }
+  /**
+   * @description Allows users to update a comment of an article.
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - It returns the request response object.
+   */
+
+  static async update(req, res) {
+    try {
+      const { body: newBody } = req.body;
+      const { id: commentId } = req.params;
+
+      if (newBody === " ") {
```

Comment:
@YvesIraguha This can be validated directly from the endpoint using Joi validator or any other express validator. It is advisable to reduce the number of branches (conditions) as much as you to avoid code complexity